### PR TITLE
LWG-3810: CTAD for `basic_format_args`

### LINKED ISSUE #3414: LWG-3810 CTAD for `std::basic_format_args`
LWG-3810 CTAD for `std::basic_format_args`

### PR DESCRIPTION
**Pull request by JMazurkiewicz**
Closes #3414.

### CODE REVIEW

File: stl/inc/format [hunk 1]
```diff
@@ -1991,6 +1991,9 @@ private:
     const _Format_arg_index* _Index_array = nullptr;
 };
 
+template <class _Context, class... _Args>
+basic_format_args(_Format_arg_store<_Context, _Args...>) -> basic_format_args<_Context>;
```
Comment (StephanTLavavej) [inline stl/inc/format:1995]: No change requested, note to other reviewers: Deduction guides can't be exported for modules, they're just automatically available if the class is exported (which is the case here), so the lack of any `_EXPORT_STD` is normal.

**Review by StephanTLavavej — APPROVED** [review]

File: tests/std/tests/P0645R10_text_formatting_args/test.cpp [hunk 2]
```diff
@@ -229,11 +229,26 @@ void test_visit_monostate() {
...
+constexpr bool verify_lwg3810 = false;
+
+template <class Context>
+constexpr bool verify_lwg3810<Context, basic_format_args<Context>> = true;
+
+template <class Context>
+void test_lwg3810() {
+    auto args_store                         = make_format_args<Context>(1, 2, 3);
+    [[maybe_unused]] basic_format_args args = args_store;
+    static_assert(verify_lwg3810<Context, decltype(args)>);
```
Comment (CaseyCarter) [inline tests/std/tests/P0645R10_text_formatting_args/test.cpp:242]: `static_assert(same_as<decltype(args), basic_format_args<Context>>);` isn't much longer, and would save the trouble of defining `verify_lwg3810`. `static_assert(same_as<decltype(basic_format_args(args_store)), basic_format_args<Context>>);` would save another line, and make it painfully obvious we're testing CTAD. (Pure style comment, ignore if you like.)

**Review by CaseyCarter — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I've pushed a commit to address @CaseyCarter's suggestion (using braces for the temporary, and applying `[[maybe_unused]]` to `args_store` even though compilers don't warn yet that it's unused at runtime). What persuaded me is that now we don't have to reason about the behavior of a test-only variable template. (Fewer lines is nicer but not decisive.)

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for implementing more Cat Template Argument Destruction, where cats bat fragile template arguments off the table to smash on the floor! :cat2: :feet: :smirk_cat:

**Pull request merged**